src/builtins.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::HashMap;
+
+use inkwell::values::{BasicValue, BasicValueEnum};
+use lazy_static::lazy_static;
+
+use crate::{
+    ast::{AstStatement, CompilationUnit, LinkageType, SourceRange},
+    codegen::generators::expression_generator::ExpressionCodeGenerator,
+    diagnostics::Diagnostic,
+    lexer::{self, IdProvider},
+    parser,
+};
+
+// Defines a set of functions that are always included in a compiled application
+lazy_static! {
+    static ref BUILTIN: HashMap<&'static str, BuiltIn> = HashMap::from([
+        (
+            "ADR",
+            BuiltIn {
+                decl: "FUNCTION ADR<T: ANY> : LWORD
+                VAR_INPUT
+                    in : T;
+                END_VAR
+                END_FUNCTION
+            ",
+                code: |generator, params, location| {
+                    if let [reference] = params {
+                        generator
+                            .generate_element_pointer(reference)
+                            .map(|it| generator.ptr_as_value(it))
+                    } else {
+                        Err(Diagnostic::codegen_error(
+                            "Expected exadtly one parameter for REF",
+                            location,
+                        ))
+                    }
+                }
+            },
+        ),
+        (
+            "REF",
+            BuiltIn {
+                decl: "FUNCTION REF<T: ANY> : REF_TO T
+                VAR_INPUT
+                    in : T;
+                END_VAR
+                END_FUNCTION
+                ",
+                code: |generator, params, location| {
+                    if let [reference] = params {
+                        generator
+                            .generate_element_pointer(reference)
+                            .map(|it| it.as_basic_value_enum())
+                    } else {
+                        Err(Diagnostic::codegen_error(
+                            "Expected exadtly one parameter for REF",
+                            location,
+                        ))
+                    }
+                }
+            },
+        )
+    ]);
+}
+
+pub struct BuiltIn {
+    decl: &'static str,
```

Comment:
Honestly, I don't remember, it feels like less code (not having .to_string() in each entry)
I think initially this came when I was using a static array of declarations instead of the lazy_static map